flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/spargel/VertexCentricIteration.java

```diff
@@ -228,48 +213,92 @@ public void setInput(DataSet<Vertex<VertexKey, VertexValue>> inputData) {
 	{
 		return new VertexCentricIteration<VertexKey, VertexValue, Message, EdgeValue>(uf, mf, edgesWithValue, maximumNumberOfIterations);
 	}
-	
+
+	/**
+	 * Configures this vertex-centric iteration with the provided parameters.
+	 *
+	 * @param parameters the configuration parameters
+	 */
+	public void configure(IterationConfiguration parameters) {
+		this.configuration = parameters;
+	}
+
+	/**
+	 * @return the configuration parameters of this vertex-centric iteration
+	 */
+	public IterationConfiguration getIterationConfiguration() {
+		return this.configuration;
+	}
+
 	// --------------------------------------------------------------------------------------------
 	//  Wrapping UDFs
 	// --------------------------------------------------------------------------------------------
 	
-	private static final class VertexUpdateUdf<VertexKey extends Comparable<VertexKey> & Serializable, 
-		VertexValue extends Serializable, Message> 
-		extends RichCoGroupFunction<Tuple2<VertexKey, Message>, Vertex<VertexKey, VertexValue>, Vertex<VertexKey, VertexValue>>
-		implements ResultTypeQueryable<Vertex<VertexKey, VertexValue>>
+	private static abstract class VertexUpdateUdf<VertexKey extends Comparable<VertexKey> & Serializable,
+		VV extends Serializable, VertexValue extends Serializable, Message>
```

Comment:
If we change VertexValue type to VV, we should be consistent and change it throughout the class. We could also simplify VertexKey to K and EdgeValue to EV to be consistent with the rest of the classes in Gelly.